Even more random events

wersal454 opened the pull request:
## What type of PR is this?
1. [ ] Bug
2. [x] Change
3. [x] Enhancement
4. [ ] Miscellaneous

### What have you changed and why?
Information:

Added new events and changed behavior of old one (a tiny bit)

+ Police skirmish event - police group(s) fight invaders group(s) in city near frontline
 + Police skirmish against rivals, same as above but it can be any city, not only on frontline
+ Post ambush convoy, same as regular one, but with a convoy
+ Post battle - remains of a freshly fought battle with injured and deceased solders from both sides, side that "won" has a few patrolling solders
+ Repair event - some enemy vehicle broke down, repair vehicle is sent to assist, after repair, both go back to closest base
+ Medevac event - kinda similar to post ambush event, but crew is alive, but injured , medevac vehicle is sent (can be transport helicopter) to pick up injured and deliver them to base. Medevac and crew are set to neutral and are unarmed, so... don't shoot them pretty pls
+ SpecOps airdrop - spawns a spec ops squad near control marker (where specops or patrolling snipers would spawn) and sends transport plane to drop light vehicle, ammobox or supplies 

old ones

+ modified helislingload event so it can to also carry cargo to spec ops 
+ some visual changes to regular post ambush event

Also I added reroll to all event functions (previously it would simply cancel event roll and would have to wait until next tick)

### Please specify which Issue this PR Resolves (If Applicable).
"This PR closes #XXXX!"

### Please verify the following.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?

1. [ ] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps:

********************************************************
Notes:

wersal454 committed:
Update fn_encounter_selectAndExecuteEvent.sqf

wersal454 committed:
new events

wersal454 committed:
comment translation

wersal454 committed:
frontline check

wersal454 committed:
Merge branch 'unstable' into patch-65

UnseenKill committed:
Merge branch 'unstable' into patch-65

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf:
```diff
@@ -0,0 +1,250 @@
+#include "..\defines.inc"
```
Need to also include `Constants.inc` in all these files where you're using defines from there

wersal454 committed:
Merge branch 'unstable' into patch-65

wersal454 committed:
Constants.inc inclusion

wersal454 committed:
Merge branch 'patch-65' of https://github.com/wersal454/A3-Antistasi into patch-65

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_HeliSlingloadCargo.sqf:
```diff
@@ -64,9 +74,8 @@ private _heliVehicleData = [_actualspawnPosition, 90, _HeliClass, _side] call A3
-    isEventInProgress = false;
-    publicVariableServer "isEventInProgress";
+    Info("Your heli can't do shit, rerolling.");
+    [VEH_SLINGLOADTRANSPORT] remoteExecCall ["SCRT_fnc_encounter_selectAndExecuteEvent", 2];
```
Probably unlikely, but could cause infinite looping or re-rolling many times id there's not some kind of counter.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_civHeli.sqf:
```diff
@@ -25,9 +26,8 @@ if (_lowCiv || _civNonHuman) exitWith {
-    isEventInProgress = false;
-    publicVariableServer "isEventInProgress";
+    Error("No civ heli found, rerolling.");
+    [CIV_HELI] remoteExecCall ["SCRT_fnc_encounter_selectAndExecuteEvent", 2];
```
Same comment here, could use a counter or something similar.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_civHeli.sqf:
```diff
@@ -54,7 +54,7 @@ private _heliGroup = _civHeliData select 2;
 _vehicles pushBack _heliVeh;
 
-_heliVeh flyInHeight (100 + (random 150));
+_heliVeh flyInHeight (60 + (random 150));
```
Seems pretty low for a CIV heli, but doesn't really matter

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_civPlane.sqf:
```diff
@@ -25,9 +26,8 @@ if (_lowCiv || _civNonHuman) exitWith {
-    isEventInProgress = false;
-    publicVariableServer "isEventInProgress";
+    Error("No civ plane found, rerolling.");
+    [CIV_PLANE] remoteExecCall ["SCRT_fnc_encounter_selectAndExecuteEvent", 2];
```
Same comment here, could use a counter or something similar.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_civPlane.sqf:
```diff
@@ -48,7 +48,7 @@ private _planeGroup = _civPlaneData select 2;
 _vehicles pushBack _planeVeh;
 
-private _height = 550 + (random 150);
+private _height = 100 + (random 150);
```
This is *extremely* low for a civilian plane

wersal454 replied:
I guess, I could make it 150. The whole point of this and heli event is to see them. Won't be able to see much at 550 (plus 150).

jwoodruff40 replied:
Fair enough. 150 works.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_fleeingCivConvoy.sqf:
```diff
@@ -29,10 +30,9 @@ Info_2("%1 will be used as center of the event at %2 position.", name _player, s
-    Info("No cities in proximity, aborting civ convoy Event.");
-    isEventInProgress = false;
-    publicVariableServer "isEventInProgress";
+if (_cities isEqualTo []) exitWith {
```
`_cities` is not defined

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_fleeingCivConvoy.sqf:
```diff
@@ -29,10 +30,9 @@ Info_2("%1 will be used as center of the event at %2 position.", name _player, s
-    publicVariableServer "isEventInProgress";
+if (_cities isEqualTo []) exitWith {
+    Info("No cities in proximity, rerolling another event.");
+    [CIV_CONVOY] remoteExecCall ["SCRT_fnc_encounter_selectAndExecuteEvent", 2];
```
Same comment here, could use a counter or something similar.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_frontlineSkirmish.sqf:
```diff
@@ -17,32 +18,19 @@ if (isNil "_player") exitWith {
-    isEventInProgress = false;
-    publicVariableServer "isEventInProgress";
+    Info("No outposts in proximity, aborting Skirmish fronline Event and rerolling for another.");
+    [SKIRMISH_FRONTLINE] remoteExecCall ["SCRT_fnc_encounter_selectAndExecuteEvent", 2];
```
Same comment here, could use a counter or something similar.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_frontlineSkirmish.sqf:
```diff
@@ -17,32 +18,19 @@ if (isNil "_player") exitWith {
 private _faction = Faction(_side);
 private _faction2 = Faction(_side2);
-private _FrontlineOutpostPosition = _originPosition ;//getMarkerPos _FrontlineOutpost;
+private _FrontlineOutpostPosition = _originPosition;
```
Why are we putting the skirmish position on the player position?

wersal454 replied:
Good question. I think I forgot to change it back during testing. Might aswell delete `_originPosition;` since it not used here.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_vehicleMove.sqf:
```diff
@@ -51,8 +52,7 @@ private _truckClass = selectRandom ((_faction get "vehiclesAmmoTrucks") + (_fact
     Error("No trucks, problems with template, aborting Vehicle Move Event.");
-    isEventInProgress = false;
-    publicVariableServer "isEventInProgress";
+    [VEH_MOVE] remoteExecCall ["SCRT_fnc_encounter_selectAndExecuteEvent", 2];
```
Same comment here, could use a counter or something similar.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_postAmbush.sqf:
```diff
@@ -15,6 +16,13 @@ if (isNil "_player") exitWith {
+
+if !(_frontLine isEqualTo []) exitWith {
+    Error("Position is near frontline, need to select appropriate event.");
+    [VEH_POSTAMBUSH] remoteExecCall ["SCRT_fnc_encounter_selectAndExecuteEvent", 2];
```
Same comment here, could use a counter or something similar.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_postAmbush.sqf:
```diff
@@ -38,12 +46,16 @@ _crater setDir _dirveh;
 private _faction = Faction(_side);
 
-private _isFia = if (random 10 > (tierWar + difficultyCoef)) then {true} else {false};
+private _isFia = if (random 10 > tierWar) then {true} else {false};
```
```suggestion
private _isFia = random 10 > tierWar;
```

Why remove the difficulty coeff?

wersal454 replied:
because...make it a little more reliable? Higher war level, more higher tier equipment is used, higher chance of it being wasted in the ambush...kinda?

jwoodruff40 replied:
Alright, fair enough.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_vehiclePatrol.sqf:
```diff
@@ -45,15 +54,14 @@ private _faction = Faction(_side);
     Info("No vehicle class, aborting Vehicle Patrol Event.");
-    isEventInProgress = false;
-    publicVariableServer "isEventInProgress";
+    [VEH_PATROL] remoteExecCall ["SCRT_fnc_encounter_selectAndExecuteEvent", 2];
```
Same comment here, could use a counter or something similar.

jwoodruff40 commented on A3A/addons/scrt/Rivals/fn_rivals_encounter_rovingMortar.sqf:
```diff
@@ -178,6 +178,7 @@ private _minSleepTime = (1 - ((5 - inactivityLevelRivals) + 1) * 0.1) * _setupTi
 private _sleepTime = _minSleepTime + random (_setupTime - _minSleepTime);
 
 _mortarGroup deleteGroupWhenEmpty true;
+sleep 10;
```
Again I don't think this does anything because it's run unscheduled

jwoodruff40 commented on A3A/addons/scrt/Rivals/fn_rivals_encounter_policeVsRivalsskirmish.sqf:
```diff
@@ -0,0 +1,172 @@
+    publicVariableServer "isEventInProgress";
+};
+
+private _cities = citiesX select {sidesX getVariable [_x, sideUnknown] != teamPlayer && {spawner getVariable _x != 2}};
```
If we can, should probably filter the city list to those that are within occpuier territory or close to an occupier outpost of some variety. Would be weird to get an occupants vs rivals skirmish in a city surrounded by invaders.

wersal454 replied:
mmm, maybe? But then again, invaders might aswell send pesky rivals to fight police instead of wasting regular troops.

jwoodruff40 replied:
I mean I guess the way you have it is more consistent with the rest of the events. Another question though: does this filter out cities that have been destroyed by invaders? It definitely wouldn't make sense to have occupant police still in a destroyed city.

wersal454 replied:
good question... not sure.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_policeSkirmish.sqf:
```diff
@@ -0,0 +1,178 @@
+
+if (_cities isEqualTo []) exitWith {
+    Info("No neutral cities available, aborting event.");
+    [POLICE_SKIRMISH] remoteExecCall ["SCRT_fnc_encounter_selectAndExecuteEvent", 2];
```
Same comment here, could use a counter or something similar.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_policeSkirmish.sqf:
```diff
@@ -0,0 +1,178 @@
+if (_frontLine isEqualTo []) exitWith {
+    Error("Position is not near frontline, need to select appropriate event.");
+    [POLICE_SKIRMISH] remoteExecCall ["SCRT_fnc_encounter_selectAndExecuteEvent", 2];
+};
```
not used anywhere
```suggestion
```

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf:
```diff
@@ -0,0 +1,251 @@
+    isEventInProgress = false;
+    publicVariableServer "isEventInProgress";
+}; */
+
```
```suggestion
```

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf:
```diff
@@ -0,0 +1,251 @@
+
+if (count _road == 0) exitWith {
+    Error("No suitable road found for medevac, rerolling.");
+    [VEH_MEDEVAC] remoteExecCall ["SCRT_fnc_encounter_selectAndExecuteEvent", 2];
```
Same comment here, could use a counter or something similar.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf:
```diff
@@ -0,0 +1,251 @@
+private _side = sidesX getVariable [_marker, Occupants];
+private _faction = Faction(_side);
+
+private _isFia = if (random 10 > tierWar) then {true} else {false};
```
```suggestion
private _isFia = random 10 > tierWar;
```

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf:
```diff
@@ -0,0 +1,251 @@
+private _MedicalVehicleClass = selectRandom ((_faction get "vehiclesMedical") + (_faction get "vehiclesHelisTransport")); // Get a random medical vehicle class
+
+if (_MedicalVehicleClass == "") exitWith {
+    Error("No vehicle class, aborting.");
```
```suggestion
    Error("No medical vehicle class, aborting.");
```

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf:
```diff
@@ -0,0 +1,251 @@
+};
+
+private _positionCrashedVehicle = getPos _crashedVehicle;
+private _flaer = "F_40mm_Red" createVehicle _positionCrashedVehicle;
```
```suggestion
private _flare = "F_40mm_Red" createVehicle _positionCrashedVehicle;
_others pushBack _flare;
```

wersal454 replied:
_flare not `_flare_` I think. 
<img width="460" height="53" alt="image" src="https://github.com/user-attachments/assets/9388d5a5-7636-42f2-88fe-1b79cc69ed1e" />

jwoodruff40 replied:
yep, my mistake. fixed

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf:
```diff
@@ -0,0 +1,251 @@
+private _flaer = "F_40mm_Red" createVehicle _positionCrashedVehicle;
+private _smokeGrenade = "SmokeShellRed" createVehicle _positionCrashedVehicle;
+
+_flaer setPos [(_positionCrashedVehicle select 0) - 4, (_positionCrashedVehicle select 1) + 4, 0];
```
```suggestion
_flare setPos [(_positionCrashedVehicle select 0) - 4, (_positionCrashedVehicle select 1) + 4, 0];
```

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf:
```diff
@@ -0,0 +1,251 @@
+
+private _positionCrashedVehicle = getPos _crashedVehicle;
+private _flaer = "F_40mm_Red" createVehicle _positionCrashedVehicle;
+private _smokeGrenade = "SmokeShellRed" createVehicle _positionCrashedVehicle;
```
```suggestion
private _smokeGrenade = "SmokeShellRed" createVehicle _positionCrashedVehicle;
_others pushBack _smokeGrenade;
```

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_repair.sqf:
```diff
@@ -0,0 +1,225 @@
+};
+
+private _originPosition = position _player;
+
+private _frontLine = (outposts + milbases + airportsX + resourcesX + factories + citiesX) select {([_x] call A3A_fnc_isFrontlineNoFIA && {sidesX getVariable [_x,sideUnknown] != teamPlayer})};
+
+if !(_frontLine isEqualTo []) exitWith {
+    Error("Position is near frontline, need to select appropriate event.");
+    [VEH_REPAIR] remoteExecCall ["SCRT_fnc_encounter_selectAndExecuteEvent", 2];
```
Same comment here, could use a counter or something similar.

jwoodruff40 replied:
same thing here. I'm nowhere close to the frontline and getting this error no matter where I go. I think this check needs to be changed.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_policeSkirmish.sqf:
```diff
@@ -0,0 +1,178 @@
+
+if (_frontLine isEqualTo []) exitWith {
+    Error("Position is not near frontline, need to select appropriate event.");
+    [POLICE_SKIRMISH] remoteExecCall ["SCRT_fnc_encounter_selectAndExecuteEvent", 2];
```
Same comment here, could use a counter or something similar.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_repair.sqf:
```diff
@@ -0,0 +1,225 @@
+private _side = sidesX getVariable [_marker, Occupants];
+private _faction = Faction(_side);
+
+private _isFia = if (random 10 > tierWar) then {true} else {false};
```
```suggestion
private _isFia = random 10 > tierWar;
```

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_repair.sqf:
```diff
@@ -0,0 +1,225 @@
+private _repairVehicleClass = selectRandom ((_faction get "vehiclesRepairTrucks")); // Get random repair vehicle
+
+if (_repairVehicleClass == "") exitWith {
+    Error("No vehicle class, aborting.");
```
```suggestion
    Error("No repair vehicle class, aborting.");
```

wersal454 committed:
Merge branch 'unstable' into patch-65

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_postBattle.sqf:
```diff
@@ -0,0 +1,465 @@
+
+if (_frontlineSitesNearPlayer isEqualTo []) exitWith {
+    Info("No outposts in proximity, aborting post battle Event.");
+    [VEH_POSTBATTLE] remoteExecCall ["SCRT_fnc_encounter_selectAndExecuteEvent", 2];
```
Same comment here, could use a counter or something similar.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf:
```diff
@@ -0,0 +1,251 @@
+
+Info("medical Event Init.");
+
+private _vehicles = [];
+private _groups = [];
```
```suggestion
private _vehicles = [];
private _groups = [];
private _others = [];
```

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf:
```diff
@@ -0,0 +1,251 @@
+    }}
+};
+
+{[_x] spawn A3A_fnc_vehDespawner} forEach _vehicles;
+{[_x] spawn A3A_fnc_groupDespawner} forEach _groups;
```
```suggestion
{[_x] spawn A3A_fnc_vehDespawner} forEach _vehicles;
{[_x] spawn A3A_fnc_groupDespawner} forEach _groups;
{deleteVehicle _x} forEach _others;
```

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_postAmbushConvoy.sqf:
```diff
@@ -0,0 +1,330 @@
+
+if !(_frontLine isEqualTo []) exitWith {
+    Error("Position is near frontline, need to select appropriate event.");
+    [VEH_POSTAMBUSHCONVOY] remoteExecCall ["SCRT_fnc_encounter_selectAndExecuteEvent", 2];
```
Same comment here, could use a counter or something similar.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_postAmbushConvoy.sqf:
```diff
@@ -0,0 +1,330 @@
+
+Info_2("%1 will be used as center of the event at %2 position.", name _player, str _originPosition);
+
+private _spawnPosition = [_originPosition, 900, distanceSPWN, 0, 0] call BIS_fnc_findSafePos;
+private _road = objNull;
+private _radiusX = 5;
+
+while {true} do {
+    _road = _spawnPosition nearRoads _radiusX;
+    if (count _road > 1 && {_road findIf {(position _x) distance2D _originPosition > 500} != -1}) exitWith {};
+    _radiusX = _radiusX + 5;
+};
+
+private _mainRoad = (_road select {(position _x) distance2D _originPosition > 500}) select 0;
+private _roadcon = roadsConnectedto _mainRoad;
+private _dirveh = if (count _roadcon > 0) then {
+    [_mainRoad, _roadcon select 0] call BIS_fnc_dirTo; // Direction from main segment to first connected road
+} else {
+    random 360; // Fallback direction if no connected roads
+};
+
+private _roadSegments = [_mainRoad];
+private _connectedRoads = _roadcon;
+
+while {count _roadSegments < 3 && count _connectedRoads > 0} do {
+    private _nextSegment = _connectedRoads select 0;
+    if (!(_nextSegment in _roadSegments)) then {
+        _roadSegments pushBack _nextSegment;
+        _connectedRoads = roadsConnectedto _nextSegment;
+    };
+};
```
This code (or very similar) is used *a lot*. Perhaps make a dedicated function for this to make reading these functions easier and less redundant?

wersal454 replied:
Is it? I think code with road segments is used only here.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_postAmbushConvoy.sqf:
```diff
@@ -0,0 +1,330 @@
+        _roadPosition = _roadPosition getPos [_latOffset, _roadDir + 90];
+    };
+
+    private _isFia = if (random 10 > tierWar) then {true} else {false};
```
```suggestion
    private _isFia = random 10 > tierWar;
```

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_postAmbushConvoy.sqf:
```diff
@@ -0,0 +1,330 @@
+    
+    private _groupCrew = createGroup _side;
+    private _crewCount = switch (true) do {
+        case (_vehicleClass in ((_faction get "vehiclesAPCs") + (_faction get "vehiclesIFVs") + (_faction get "vehiclesLightTanks") + (_faction get "vehiclesMilitiaAPCs") + (_faction get "vehiclesAA"))): { 3 };
```
```suggestion
        case (_vehicleClass in (_faction get "vehiclesArmor")): { 3 };
```

wersal454 replied:
Are all of those in `vehiclesArmored` ?

jwoodruff40 replied:
Sorry, should have been `vehiclesArmor`. Fixed the suggestion. See our discussion on discord.

wersal454 replied:
Ah, yeah, so, militia apcs are not there (maybe add them?) and chance to spawn artillery I think is a bit much?

jwoodruff40 replied:
It's up to you.

You could do vehiclesArmor + vehiclesMilitiaAPCs or just leave it as is.

wersal454 replied:
Then I guess it's better to leave as is.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_policeHostages.sqf:
```diff
@@ -0,0 +1,114 @@
+
+if (_cities isEqualTo []) exitWith {
+    Info("No neutral cities available, aborting event.");
+    [POLICE_HOSTAGE] remoteExecCall ["SCRT_fnc_encounter_selectAndExecuteEvent", 2];
```
Same comment here, could use a counter or something similar.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_SpecOpsAirdrop.sqf:
```diff
@@ -0,0 +1,214 @@
+private _outpostPosition = getMarkerPos _outpost;
+
+if (_outpostPosition isEqualTo []) exitWith {
+    Info("No spawn positions in proximity, aborting Heli slingload cargo Event.");
```
```suggestion
    Info("No spawn positions in proximity, aborting SpecOps Airdrop Event.");
```

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_SpecOpsAirdrop.sqf:
```diff
@@ -0,0 +1,214 @@
+_AirDropPositionActuall = [_AirDropPosition, 40, 100, 10, 0, 5, 0, [], [[0,0,0],[0,0,0]]] call BIS_fnc_findSafePos;
+
+private _specOpsArray = if (_difficult) then {selectRandom (_faction get "groupSpecOpsRandom")} else {selectRandom ([_faction, "groupsTierSquads"] call SCRT_fnc_unit_flattenTier)}; 
+_InfGroup = [_AirDropPositionActuall, _side, _specOpsArray] call A3A_fnc_spawnGroup;
```
```suggestion
private _InfGroup = [_AirDropPositionActuall, _side, _specOpsArray] call A3A_fnc_spawnGroup;
```

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_SpecOpsAirdrop.sqf:
```diff
@@ -0,0 +1,214 @@
+
+private _smokeGrenade = "SmokeShellRed" createVehicle _AirDropPosition;
+
+private _smokes = _AirDropPosition nearObjects ["SmokeShell", 200];
```
```suggestion
private _smokes = _AirDropPosition nearObjects ["SmokeShell", 200];
_others append _smokes;
```

wersal454 replied:
Aren't smokes delete themselves after they run out? If they do, maybe just delete `_others` since it's not used?

jwoodruff40 replied:
I'm not sure, would need to test. I think it depends how they're spawned. Maybe since you're using the built in smokeshell instead of creating the effects manually, they do?

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_SpecOpsAirdrop.sqf:
```diff
@@ -0,0 +1,214 @@
+    private _smokes = _AirDropPosition nearObjects ["SmokeShell", 200];
+    private _dropPosition = _AirDropPosition;
+
+    if(count _smokes > 0) then {
```
Isn't this redundant?

wersal454 replied:
it is... this reminds me, I think I forgot to add chemlight

wersal454 replied:
<img width="1191" height="1066" alt="image" src="https://github.com/user-attachments/assets/593c9b5b-27f8-4582-a1b5-4f60bd1055fa" />

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_SpecOpsAirdrop.sqf:
```diff
@@ -0,0 +1,214 @@
+    sleep 1;
+    private _box1 = [_boxType,position _planeVeh] call SCRT_fnc_common_airdropCargo;
+    _box1 enableRopeAttach true;
+    _box1 allowDamage false;
```
even if it's a vehicle? So we're dropping in an unkillable vehicle?

wersal454 replied:
Unless it lands (no?). Similar logic with helislingload.

jwoodruff40 replied:
Can we set `allowDamage true` if it's a vehicle and successfully lands then?

wersal454 replied:
oh. I forgot to enable damage for both box1 and box1

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_SpecOpsAirdrop.sqf:
```diff
@@ -0,0 +1,214 @@
+	}
+};
+
+{[_x] spawn A3A_fnc_vehDespawner} forEach _vehicles;
+{[_x] spawn A3A_fnc_groupDespawner} forEach _groups;
```
```suggestion
{[_x] spawn A3A_fnc_vehDespawner} forEach _vehicles;
{[_x] spawn A3A_fnc_groupDespawner} forEach _groups;
{_deleteVehicle _x} forEach _others;
```

wersal454 replied:
wouldn't it be better to put `_others` in `_vehicles` and use despawner?

jwoodruff40 replied:
¯⁠\⁠_⁠(⁠ツ⁠)⁠_⁠/⁠¯

wersal454 replied:
decided to put others in vehicles and use despawner

wersal454 commented:
Concerning rerolls, Iirc it will put "rejected" events on ice, so they won't be put in the loop (so it won't try infinite times.). Eventually it will land on some event that will 100% happen. I think I discussed this with Silence.

wersal454 committed:
Update A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf

Co-authored-by: John Woodruff <22091381+jwoodruff40@users.noreply.github.com>
Signed-off-by: wersal <wroma@bk.ru>

wersal454 committed:
Update A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf

Co-authored-by: John Woodruff <22091381+jwoodruff40@users.noreply.github.com>
Signed-off-by: wersal <wroma@bk.ru>

jwoodruff40 commented:
> Concerning rerolls, Iirc it will put "rejected" events on ice, so they won't be put in the loop (so it won't try infinite times.). Eventually it will land on some event that will 100% happen. I think I discussed this with Silence.

Ok, I mis-read selectAndExecuteEvent. I get it now. Will remove those comments.

wersal454 committed:
Update A3A/addons/scrt/Encounter/fn_encounter_SpecOpsAirdrop.sqf

Co-authored-by: John Woodruff <22091381+jwoodruff40@users.noreply.github.com>
Signed-off-by: wersal <wroma@bk.ru>

wersal454 committed:
Update A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf

Co-authored-by: John Woodruff <22091381+jwoodruff40@users.noreply.github.com>
Signed-off-by: wersal <wroma@bk.ru>

wersal454 committed:
Update A3A/addons/scrt/Encounter/fn_encounter_SpecOpsAirdrop.sqf

Co-authored-by: John Woodruff <22091381+jwoodruff40@users.noreply.github.com>
Signed-off-by: wersal <wroma@bk.ru>

wersal454 committed:
Update A3A/addons/scrt/Encounter/fn_encounter_repair.sqf

Co-authored-by: John Woodruff <22091381+jwoodruff40@users.noreply.github.com>
Signed-off-by: wersal <wroma@bk.ru>

wersal454 committed:
Update A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf

Co-authored-by: John Woodruff <22091381+jwoodruff40@users.noreply.github.com>
Signed-off-by: wersal <wroma@bk.ru>

wersal454 committed:
Update A3A/addons/scrt/Encounter/fn_encounter_postAmbush.sqf

Co-authored-by: John Woodruff <22091381+jwoodruff40@users.noreply.github.com>
Signed-off-by: wersal <wroma@bk.ru>

wersal454 committed:
Update A3A/addons/scrt/Encounter/fn_encounter_repair.sqf

Co-authored-by: John Woodruff <22091381+jwoodruff40@users.noreply.github.com>
Signed-off-by: wersal <wroma@bk.ru>

wersal454 committed:
Update A3A/addons/scrt/Encounter/fn_encounter_postAmbushConvoy.sqf

Co-authored-by: John Woodruff <22091381+jwoodruff40@users.noreply.github.com>
Signed-off-by: wersal <wroma@bk.ru>

wersal454 committed:
specopsairdrop fixes

wersal454 committed:
_cities not definced

wersal454 committed:
only non destroyed cities

untested

wersal454 committed:
higher hight

wersal454 committed:
proper skirmish position

wersal454 committed:
Update A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf

Co-authored-by: John Woodruff <22091381+jwoodruff40@users.noreply.github.com>
Signed-off-by: wersal <wroma@bk.ru>

wersal454 committed:
medical cleanup

wersal454 committed:
Update A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf

Co-authored-by: John Woodruff <22091381+jwoodruff40@users.noreply.github.com>
Signed-off-by: wersal <wroma@bk.ru>

wersal454 committed:
Merge branch 'patch-65' of https://github.com/wersal454/A3-Antistasi into patch-65

wersal454 commented:
Added all fixes

jwoodruff40 reviewed (changes requested):
- `POLICE_HOSTAGE` missing from Constants.inc
- encounter_MedEvac acts strangely.
  - flare and smoke didn't spawn until the damaged vehicle was accidentally destroyed
  - medevac helicopter flying all over the place
  - injured crew didn't seem to spawn until way after vehicle was detroyed
- specopsAirdrop causes a lot of these errors, though it mostly seems to work:
  - <img width="792" height="471" alt="image" src="https://github.com/user-attachments/assets/6726fe61-e4e3-45b4-87e8-ad8c8cc9b93c" />
- getting this for postBattle:
  - <img width="680" height="189" alt="image" src="https://github.com/user-attachments/assets/212465c0-f08e-4efb-a63d-cc16a9a36d97" />

wersal454 committed:
police hostage defined

wersal454 committed:
possible fix for heli not landing

wersal454 committed:
specops fix

wersal454 committed:
post battle roadPosition fix

wersal454 committed:
attempt to fix issues with medevac

wersal454 committed:
Merge branch 'unstable' into patch-65

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_postAmbushConvoy.sqf:
```diff
@@ -0,0 +1,330 @@
+
+private _frontLine = (outposts + milbases + airportsX + resourcesX + factories + citiesX) select {([_x] call A3A_fnc_isFrontlineNoFIA && {sidesX getVariable [_x,sideUnknown] != teamPlayer})};
+
+if !(_frontLine isEqualTo []) exitWith {
```
This seems to be filtering out all locations unless the player just happens to control all the frontline cities. Is that the intent?

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf:
```diff
@@ -0,0 +1,259 @@
+private _cityPositions = citiesX apply {getMarkerPos _x};
+private _militaryPositions = (outposts + milbases + airportsX) apply {getMarkerPos _x};
+
+while {true} do {
+    _road = _spawnPosition nearRoads _radiusX;
+    
+    if (count _road > 0 && {_road findIf {
+        private _roadPos = position _x;
+        // Main condition with added check for military bases
+        (_roadPos distance2D _originPosition > 500) &&
+        ({_roadPos distance2D _x < 700} count _cityPositions == 0) &&
+        ({_roadPos distance2D _x < 700} count _militaryPositions == 0)
+    } != -1}) exitWith {};
+    
+    _radiusX = _radiusX + 5;
+    if (_radiusX > 150) exitWith {_road = [];};
+};
+
+if (count _road == 0) exitWith {
+    Error("No suitable road found for medevac, rerolling.");
+    [VEH_MEDEVAC] remoteExecCall ["SCRT_fnc_encounter_selectAndExecuteEvent", 2];
+};
```
Tried this like 10 times and always got no suitable road found

jwoodruff40 reviewed (changes requested):
Nearly there. A few more issues.

- postBattle: the fires look kind of weird, since there's no smoke and they're just kinda randomly placed, not actually burning any objects. Maybe fires (with smoke) on or near the vehicles / rubble? Could probably be a bit smaller too.

![Image](https://github.com/user-attachments/assets/2b97b630-889f-4ea4-ac3b-52f2e17f2ba5)

wersal454 committed:
Merge branch 'unstable' into patch-65

wersal454 committed:
medevac fixes

wersal454 committed:
postambush fixes

wersal454 committed:
postbattle comment translation

wersal454 committed:
Update fn_encounter_repair.sqf

wersal454 commented:
increased the raduis for finding road in medevac, should increase the chances of succesfully starting event
also added additional check in waituntil, so if med. vehicle is Air, it will only wait for it's touchdown and not the <100 m. distance check.

Tried to add smoke to big fires in postBattle, didn't work (maybe particle overlap or smth), reduced number of big fires.

fixed logic for finding position for both post ambush events + minor fixes

same for repair, + minor adjustments to logic.

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf:
```diff
@@ -0,0 +1,260 @@
+isEventInProgress = false;
+publicVariableServer "isEventInProgress";
+
+Info("Post Ambush Vehicle Event clean up complete.");
```
```suggestion
Info("Medevac Event clean up complete.");
```

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf:
```diff
@@ -0,0 +1,260 @@
+
+private _roadPosition = getPos _selectedRoad;
+
+private _marker = [(markersX select {sidesX getVariable [_x, sideUnknown] != teamPlayer}), _originPosition] call BIS_fnc_nearestPosition;
```
Seems to work now. Minor suggestion: can we have the medevac bring the casualties to a mil location? Was a little weird when I was sitting in a city and they just dropped the casualties off 50m away from me in the middle of the city. (was an occupant city)

```suggestion
private _marker = [((outposts + milbases + airportsX) select {sidesX getVariable [_x, sideUnknown] != teamPlayer}), _originPosition] call BIS_fnc_nearestPosition;
```

jwoodruff40 commented on A3A/addons/scrt/Encounter/fn_encounter_repair.sqf:
```diff
@@ -0,0 +1,233 @@
+    }}
+};
+
+sleep 15;
```
```suggestion
sleep 60;
```

jwoodruff40 reviewed (changes requested):
Alright, final little stuff and this'll be g2g

wersal454 committed:
Update A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf

Co-authored-by: John Woodruff <22091381+jwoodruff40@users.noreply.github.com>
Signed-off-by: wersal <wroma@bk.ru>

wersal454 committed:
Update A3A/addons/scrt/Encounter/fn_encounter_MedEvac.sqf

Co-authored-by: John Woodruff <22091381+jwoodruff40@users.noreply.github.com>
Signed-off-by: wersal <wroma@bk.ru>

wersal454 committed:
Update A3A/addons/scrt/Encounter/fn_encounter_repair.sqf

Co-authored-by: John Woodruff <22091381+jwoodruff40@users.noreply.github.com>
Signed-off-by: wersal <wroma@bk.ru>

jwoodruff40 reviewed (approved):
lgtm
